src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
```

Comment:
Template is the first thing that comes to mind. I think I would like to use this in QuadVelocityController if we find the datapoints to be too noisy.